### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Koliras**
[DEMO LINK](https://Koliras.github.io/react_static-list-of-posts-js/)

### CODE REVIEW

File: src/components/CommentList/CommentList.jsx [hunk 1]
```diff
@@ -1,3 +1,10 @@
-export const CommentList = () => (
-  <>Put the list here</>
+import './CommentList.scss';
+import { CommentInfo } from '../CommentInfo';
+
+export const CommentList = ({ comments }) => (
+  <div className="CommentList">
+    {comments.map(comment => (
+      <CommentInfo comment={comment} />
```
Comment (AndriiKuz) [inline src/components/CommentList/CommentList.jsx:7]: You should add key props here

File: src/components/PostInfo/PostInfo.jsx [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+
+          <UserInfo user={user} />
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      {comments.length === 0
```
Comment (AndriiKuz) [inline src/components/PostInfo/PostInfo.jsx:26]: You can simplify this condition for example !comments.length

File: src/components/PostList/PostList.jsx [hunk 3]
```diff
@@ -1,3 +1,9 @@
-export const PostList = () => (
-  <>Put the list here</>
+import { PostInfo } from '../PostInfo';
+
+export const PostList = ({ posts }) => (
+  <div className="PostList">
+    {posts.map(post => (
+      <PostInfo post={post} />
```
Comment (AndriiKuz) [inline src/components/PostList/PostList.jsx:6]: Here also should be key props

**Review by AndriiKuz — CHANGES_REQUESTED** [review]
GJ! but you should fix some mistakes)

File: src/components/PostInfo/PostInfo.jsx (hunk 2 above)
Comment (tiserett) [reply src/components/PostInfo/PostInfo.jsx:26]: converting value to opposite is not the best practice, just use !! or Boolean instead, this way you won't receive 0 on the page, if there's no comments + you can simplify condition

comments.length === 0 -> !!comments.length

File: src/components/UserInfo/UserInfo.jsx [hunk 4]
```diff
@@ -1,3 +1,7 @@
-export const UserInfo = () => (
-  <>Put the user here</>
+import './UserInfo.scss';
+
+export const UserInfo = ({ user }) => (
+  <a className="UserInfo" href={`mailto:${user.email}`}>
+    {user.name}
+  </a>
 );
```
Comment (tiserett) [inline src/components/UserInfo/UserInfo.jsx:7]: use destructuring

**Review by tiserett — CHANGES_REQUESTED** [review]

File: src/components/PostInfo/PostInfo.jsx [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+
+          <UserInfo user={user} />
+        </p>
+      </div>
+
+      <p className="PostInfo__body">
+        {body}
+      </p>
+
+      {comments.length
```
Comment (StasSokolov) [inline src/components/PostInfo/PostInfo.jsx:26]: this condition isn't correct, replace please `comments.length` => `!!coments.length`

File: src/App.jsx [hunk 6]
```diff
@@ -1,107 +1,20 @@
...
-// import usersFromServer from './api/users.json';
+import postsFromServer from './api/posts.json';
+import commentsFromServer from './api/comments.json';
+import usersFromServer from './api/users.json';
+import { PostList } from './components/PostList';
+
+export const posts = postsFromServer.map(post => ({
+  ...post,
+  comments: commentsFromServer.filter(comment => comment.postId === post.id),
+  user: usersFromServer.find(user => user.id === post.userId),
```
Comment (StasSokolov) [inline src/App.jsx:11]: will be good to move `commentsFromServer.filter(...)` and `usersFromServer.filter(...)` into separated function

**Review by StasSokolov1 — CHANGES_REQUESTED** [review]
great job, left to fix only few comments!

File: src/components/PostInfo/PostInfo.jsx (hunk 5 above)
Comment (Koliras) [reply src/components/PostInfo/PostInfo.jsx:26]: That's ternar operator, it converts `comments.length` to boolean automatically

File: src/App.jsx [hunk 7]
```diff
@@ -1,107 +1,28 @@
...
-
-        <b data-cy="NoCommentsMessage">No comments yet</b>
-      </div>
-
-      <div className="PostInfo">
-        <div className="PostInfo__header">
-          <h3 className="PostInfo__title">
-            doloremque illum aliquid sunt
-          </h3>
+function commentsOfPost(comments, post) {
```
Comment (Moroz-Dmytro) [inline src/App.jsx:8]:
```suggestion
function postComments(comments, post) {
```

File: src/App.jsx [hunk 8]
```diff
@@ -1,107 +1,28 @@
...
+  return comments.filter(comment => comment.postId === post.id);
+}
 
-          <p>
-            {' Posted by  '}
+function authorOfPost(users, post) {
```
Comment (Moroz-Dmytro) [inline src/App.jsx:12]:
```suggestion
function postAuthor(users, post) {
```

**Review by Moroz-Dmytro — APPROVED** [review]

**Pull request closed without merging**